src/Logic.hs

```diff
@@ -634,18 +637,35 @@ unintegrateAfter pid state = case Pr.lookupPullRequest pid state of
 
 -- | If there is an integration candidate, and its integration sha matches that of the build,
 --   then update the build status for that pull request. Otherwise do nothing.
-handleBuildStatusChanged :: Sha -> BuildStatus -> ProjectState -> Action ProjectState
-handleBuildStatusChanged buildSha newStatus state = pure $
-  compose [ Pr.updatePullRequest pid setBuildStatus
+handleBuildStatusChanged :: Sha -> Context -> BuildStatus -> ProjectState -> Action ProjectState
+handleBuildStatusChanged buildSha check newStatus state = pure $
+  compose [ Pr.updatePullRequest pid (setCheckIfNeeded . setBuildStatus)
           . case newStatus of
             BuildFailed _ -> unintegrateAfter pid
             _             -> id
           | pid <- Pr.filterPullRequestsBy shouldUpdate state
           ] state
   where
+  satisfiedCheck = contextSatisfiesChecks (Pr.mandatoryChecks state) check
```

Comment:
The changes here seem quite complex for what I understood was the intended behavior: wait for all mandatory checks to get a success, then merge.

I haven't thought this through, since I just read up on this, but to me it seems there should be a simpler solution for keeping that kind of state.

Reply:
The snippet your referring to that waits for all checks happens here: https://github.com/channable/hoff/blob/efd7b8dd5b9180db0fd8451681d363e75b8e81bc/src/Logic.hs#L829-L835.

This part is slightly different and is responsible for updating each PRs state, i.e. for example commenting on PRs when builds start and such. I tried cleaning it up a bit as I think it is misplaced to do it via the filtering. I swapped it around, which simplifies the `supersedes` function so the filtering is simpler, but the updating functions have to keep track on whether they should happen or not.